core/kazoo_ast/src/fs_prop_usage.erl

```diff
@@ -0,0 +1,648 @@
+-module(fs_prop_usage).
+
+%% module for parsing ecallmgr modules, looking for Props from FreeSWITCH
+
+-export([process/0, process/1
+        ,to_header_file/0
+        ]).
+
+-include_lib("kazoo_ast/include/kz_ast.hrl").
+-include_lib("kazoo/src/kz_json.hrl").
+
+-define(DEBUG(_Fmt, _Args), 'ok').
+%% -define(DEBUG(Fmt, Args), io:format([$~, $p, $  | Fmt], [?LINE | Args])).
+
+to_header_file() ->
+    Usage = process(),
+    write_usage_to_header(Usage).
+
+write_usage_to_header(Usage) ->
+    {'ok', IO} = file:open(event_filter_filename()
+                          ,['write'
+                           ,'delayed_write'
+                           ]),
+    write_usage_to_header(Usage, IO).
+
+write_usage_to_header(Usage, IO) ->
+    EventFilters = lists:foldl(fun write_mod_usage/2, sets:new(), Usage),
+    [First|Sorted] = lists:usort(sets:to_list(EventFilters)),
+
+    file:write(IO, io_lib:format("-define(FS_EVENT_FITLERS~n       ,[~p~n", [First])),
+    lists:foreach(fun(Filter) ->
+                          file:write(IO, io_lib:format("        ,~p~n", [Filter]))
+                  end
+                 ,Sorted
+                 ),
+    file:write(IO, "        ])."),
+    'ok' = file:close(IO),
+    io:format("wrote ~s~n", [event_filter_filename()]).
```

Comment:
So the lines not using ?DEBUG are for the escript progress thingies (once this is added to the build).